File: benchmarks/python/fibonacci_tail.py
def fib_tail(n: int, a: int, b: int) -> int:
    if n == 0:
        return a
    return fib_tail(n - 1, b, a + b)

result: int = 0
for _ in range(10000):
    result = fib_tail(10000, 0, 1)
print(result)

File: benchmarks/go/fibonacci_tail.go
package main

import "fmt"

func fibTail(n, a, b uint64) uint64 {
    if n == 0 { return a }
    return fibTail(n-1, b, a+b)
}

func main() {
    var result uint64
    for i := 0; i < 10000; i++ {
        result = fibTail(10000, 0, 1)
    }
    fmt.Println(result)
}
